src/dune_rules/super_context.ml

```diff
@@ -382,7 +382,13 @@ let () =
     let* ctx = Context.DB.by_dir dir in
     let* t = find_exn (Context.name ctx) in
     let* expander = expander t ~dir in
-    Expander.expand_str expander sw |> Action_builder.evaluate_and_collect_facts >>| fst)
+    Expander.expand_str expander sw |> Action_builder.evaluate_and_collect_facts >>| fst);
+  Expander0.set_db (fun ~dir ->
```

Comment:
I'm not sure I understand why `Expander0` needs its own db. I wonder if there's a way to introduce that module without having to add more state / side effects?

Reply:
It is not its own db, it's just a forward declaration for the db in super context. I do not know how to break cycles without using forward references. Open to suggestions though